### PR TITLE
feat(Select): adds direction prop to Select

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Allows user to specify the direction of the Select's menu expansion. Adds a SelectDirection enum for explicit directions. 

**Refer to issue**: #2368

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/Select/Select.tsx [hunk 1]
```diff
@@ -22,6 +22,8 @@ export interface SelectProps
   children: React.ReactElement[];
   /** Classes applied to the root of the Select */
   className?: string;
+  /** Flaf specifying which direction the Select menu expands */
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/Select/Select.tsx]: Flaf -> Flag

Comment (kmcfaul) [reply packages/patternfly-4/react-core/src/components/Select/Select.tsx]: Haha my bad

File: packages/patternfly-4/react-core/src/components/Select/Select.md [hunk 2]
```diff
@@ -29,7 +29,8 @@ class SingleSelectInput extends React.Component {
     this.state = {
       isExpanded: false,
       selected: null,
-      isDisabled: false
+      isDisabled: false,
+      direction: 'down'
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Select/Select.md]: should be using `SelectDirection.down`

File: packages/patternfly-4/react-core/src/components/Select/Select.md [hunk 3]
```diff
@@ -61,10 +62,22 @@ class SingleSelectInput extends React.Component {
         isDisabled: checked
       })
     }
+
+    this.toggleDirection = () => {
+      if(this.state.direction === 'up') {
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Select/Select.md]:
```suggestion
      if(this.state.direction === SelectDirection.up) {
```

File: packages/patternfly-4/react-core/src/components/Select/Select.md [hunk 4]
```diff
@@ -98,6 +112,14 @@ class SingleSelectInput extends React.Component {
           id="toggle-disabled"
           name="toggle-disabled"
         />
+        <Checkbox
+          label="Expands up"
+          isChecked={direction === 'up'}
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Select/Select.md]:
```suggestion
          isChecked={direction === SelectDirection.up}
```

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx [hunk 5]
```diff
@@ -22,8 +22,8 @@ export interface SelectDemoState {
   plainTypeaheadMultiSelected: string[],
   plainTypeaheadMultiIsPlain: boolean,
   customTypeaheadMultiIsExpanded: boolean,
-  customTypeaheadMultiSelected: string[]
-
+  customTypeaheadMultiSelected: string[],
+  direction: string
```
Comment (jschuler) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx]:
```suggestion
  direction: SelectDirection.up | SelectDirection.down
```

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx [hunk 6]
```diff
@@ -91,6 +92,18 @@ export class SelectDemo extends Component<SelectDemoState> {
     <SelectOption key={11} value={ new State('North Carolina', 'NC', 'Raleigh', 1789)} />
   ];
 
+  toggleDirection = () => {
+    if(this.state.direction === 'up') {
```
Comment (jschuler) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx]:
```suggestion
    if(this.state.direction === SelectDirection.up) {
```

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx [hunk 7]
```diff
@@ -317,6 +331,14 @@ export class SelectDemo extends Component<SelectDemoState> {
             ))}
           </Select>
         </div>
+        <Checkbox
+          label="Expands up"
+          isChecked={this.state.direction === 'up'}
```
Comment (jschuler) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx]:
```suggestion
          isChecked={this.state.direction === SelectDirection.up}
```

**Pull request merged**